aws/ec2metadata/token_provider.go

```diff
@@ -48,19 +49,23 @@ func (t *tokenProvider) fetchTokenHandler(r *aws.Request) {
 	output, err := t.client.getToken(r.Context(), t.configuredTTL)
 	if err != nil {
 		// change the disabled flag on token provider to true, when error is request timeout error.
-		if requestFailureError, ok := err.(awserr.RequestFailure); ok {
-			switch requestFailureError.StatusCode() {
-			case http.StatusForbidden, http.StatusNotFound, http.StatusMethodNotAllowed:
+		if rf, ok := err.(awserr.RequestFailure); ok {
+			switch rf.StatusCode() {
+			case http.StatusForbidden,
+				http.StatusNotFound,
+				http.StatusMethodNotAllowed:
+
 				atomic.StoreUint32(&t.disabled, 1)
+
 			case http.StatusBadRequest:
-				r.Error = requestFailureError
+				r.Error = rf
 			}
 
 			// Check if request timed out while waiting for response
-			if e, ok := requestFailureError.OrigErr().(awserr.Error); ok {
-				if e.Code() == "RequestError" {
-					atomic.StoreUint32(&t.disabled, 1)
-				}
+			var re *aws.RequestSendError
+			var ce *aws.RequestCanceledError
+			if errors.As(rf, &re) || errors.As(rf, &ce) {
+				atomic.StoreUint32(&t.disabled, 1)
 			}
```

Comment:
Just from a styling perspective, whats your thoughts on using As vs. Is in this case? Seems As is more for if you need to assign the error to that type and use it, versus Is being more of a check.

Reply:
`Is` works great for equality check, but they must be exactly equal. Where as `As` works good for an error that is like, or of type of.  `Is` is more helpful for error value comparisons, not so much for type/interfaces.